plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -454,11 +458,20 @@ public long deleteDocuments(RemoteTableName remoteTableName, TupleDomain<ColumnH
 
     public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoColumnHandle> columns)
     {
+        Set<MongoColumnHandle> projectedColumns = tableHandle.getProjectedColumns();
+        checkArgument(projectedColumns.isEmpty() || projectedColumns.equals(new HashSet<>(columns)), "Columns and projected columns must be equal");
+
+        MongoCollection<Document> collection = getCollection(tableHandle.getRemoteTableName());
         Document output = new Document();
-        for (MongoColumnHandle column : columns) {
-            output.append(column.getName(), 1);
+        for (MongoColumnHandle column : getSufficientColumns(columns)) {
+            if (column.isDbRef()) {
+                // Qualified parent column name of DBRef column
```

Comment:
Could you update the comment to explain **_why_** using qualified parent name in case of DBRef column.